service/azureconfig/v1/resource/resourcegroup/resource.go

```diff
@@ -24,33 +24,43 @@ const (
 )
 
 type Config struct {
+	Logger micrologger.Logger
+
 	AzureConfig      client.AzureConfig
-	Logger           micrologger.Logger
+	Azure            key.Azure
```

Comment:
Azure comes before AzureConfig in the alphabet. Also, what is the difference between both? I think the naming is a bit unfortunate. One is a client and one is installation/host-cluster configuration as I see it.

Reply:
Yes check https://github.com/giantswarm/azure-operator/pull/157#discussion_r176372168 I'm going to remove AzureConfig in general. Just not in this PR.